sendgrid/message.py

```diff
@@ -123,7 +125,22 @@ def add_bcc(self, bcc):
                 self.add_bcc(email)
 
     def set_replyto(self, replyto):
-        self.reply_to = replyto
+        name, email = rfc822.parseaddr(replyto.replace(',', ''))
+        if name and email:
+            self.set_reply_to_name(replyto)
+        elif email:
+            self.reply_to = email
+
+    def set_reply_to_name(self, replyto):
+        if sys.version_info < (3, 0) and isinstance(replyto, unicode):
+            replyto = replyto.encode('utf-8')

```

Comment:
Should this encoding be done for all header values? If so, would it make sense to move this logic to the `set_headers` method?